Libplanet/Blockchain/Policies/BlockPolicy.cs

```diff
@@ -121,6 +121,12 @@ public class BlockPolicy<T> : IBlockPolicy<T>
             HashDigest<SHA256>? prevHash = lastBlock?.Hash;
             DateTimeOffset? prevTimestamp = lastBlock?.Timestamp;
 
+            if (nextBlock.Index == 0 && blocks.Count != 0)
```

Comment:
Can the condition `blocks.Count != 0` be `false`?  If I understand correctly `BlockChain<T>` objects cannot be empty (i.e., there is no possibility that any empty `BlockChain<T>` instance exists), right?

Reply:
Yes. But it can be empty before the genesis block is appended.

Reply:
Okay, this method can be used by the `BlockChain<T>()` constructor at an uninitialized state, right? This kind of circular dependency should be resolved by separating this validation method into two methods: one validates any kind of blocks (i.e., also applied to the genesis), and other one validates blocks having `PreviousHash` (i.e., not applied to the genesis).